c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -403,6 +403,9 @@ struct ExpandableSegment {
 #ifndef FBCODE_CAFFE2
       prop.requestedHandleTypes = CU_MEM_HANDLE_TYPE_POSIX_FILE_DESCRIPTOR;
 #endif
+      int flag = 0;
```

Comment:
could this be static?

Reply:
Not sure that is safe because it's potentially by mutated `cuDeviceGetAttribute_`?

Reply:
Sure, I mean stashing it somewhere after only getting it once, but maybe too much premature optimization

Reply:
Doesn't matter really, because actual allocations are rare and are much slower than querying properties